docs/running-on-yarn.md

```diff
@@ -441,3 +441,81 @@ If you need a reference to the proper location to put log files in the YARN so t
 - In `cluster` mode, the local directories used by the Spark executors and the Spark driver will be the local directories configured for YARN (Hadoop YARN config `yarn.nodemanager.local-dirs`). If the user specifies `spark.local.dir`, it will be ignored. In `client` mode, the Spark executors will use the local directories configured for YARN while the Spark driver will use those defined in `spark.local.dir`. This is because the Spark driver does not run on the YARN cluster in `client` mode, only the Spark executors do.
 - The `--files` and `--archives` options support specifying file names with the # similar to Hadoop. For example you can specify: `--files localtest.txt#appSees.txt` and this will upload the file you have locally named `localtest.txt` into HDFS but this will be linked to by the name `appSees.txt`, and your application should use the name as `appSees.txt` to reference it when running on YARN.
 - The `--jars` option allows the `SparkContext.addJar` function to work if you are using it with local files and running in `cluster` mode. It does not need to be used if you are using it with HDFS, HTTP, HTTPS, or FTP files.
+
+# Running in a secure YARN cluster
+
+As covered in [security](security.html), Kerberos is used in a secure YARN cluster to
+authenticate principals with services —and so access these services and
+their data. These services issue 'hadoop tokens' to grant access to the service and data,
+tokens which are then supplied over Hadoop IPC and REST/Web APIs as proof of access rights.
+For YARN applications to interact with HDFS, HBase and Hive, the application must request tokens
+using the kerberos credentials of the user launching the cluster (the principal).
+
+This is normally done at launch time: Spark will automatically obtain a token for the cluster's
+HDFS filesystem, and optionally HBase and Hive.
+
+If an application needs to interact with other secure HDFS clusters, then
+the tokens needed to access these clusters must be explicitly requested at
+launch time. This is done by listing them in the `spark.yarn.access.namenodes` property.
+
+```
+spark.yarn.access.namenodes hdfs://ireland.emea.example.org:8020/,hdfs://frankfurt.emea.example.org:8020/
+```
+
+Hadoop tokens expire. They can be renewed "for a while"; the Spark Application Master will automatically
```

Comment:
I believe he was mistaken. He was probably thinking of the other feature where YARN, in like 2.7 or 2.8, will actually get you an hdfs token for log aggregation.  The RM will renew tokens submitted as long as they have a renewer associated.  That is hdfs, hive, hbase, MR history, etc.

But I'm good with what you propose and just get rid of who does it.